### PR TITLE
Scapy.commands improvement + doc

### PR DESCRIPTION
**Pull request by gpotter2**
This PR:
- fixes `lsc()` printing by adding extra docs to the commands listed
- improves a few of those commands / remove duplicated code
- fixes a small bug in `__iterlen__` that could result on some layers (e.g. DHCP) to the `sr()` suite never ending

fixes https://github.com/secdev/scapy/issues/1478

### CODE REVIEW

**Comment by gpotter2** [comment]
@guedou @p-l- Ready for review

File: test/regression.uts [hunk 1]
```diff
@@ -277,7 +277,7 @@ finally:
 = Test linehexdump function
 conf_color_theme = conf.color_theme
 conf.color_theme = BlackAndWhite()
-assert(linehexdump(Ether(src="00:01:02:03:04:05"), dump=True) == "FFFFFFFFFFFF0001020304059000 ..............")
+assert linehexdump(Ether(src="00:01:02:03:04:05"), dump=True) == 'ff ff ff ff ff ff 00 01 02 03 04 05 90 00  ..............'
```
Comment (guedou) [inline test/regression.uts]: Looks like a regression. Why are you changing the output?

**Comment by gpotter2** [comment]
As quoted in #1478, we have recently changed the output compared on how it was before (and how it is supposed to be, to look like tcpdump)

```
gpotter@louarn:~/Bureau/scapy-2.3.2$ ./run_scapy 
INFO: Can't import python gnuplot wrapper . Won't be able to plot.
WARNING: No route found for IPv6 destination :: (no default route?)
Welcome to Scapy (2.3.2)
>>> linehexdump(Ether(src="00:01:02:03:04:05"))
WARNING: Mac address to reach destination not found. Using broadcast.
FF FF FF FF FF FF 00 01 02 03 04 05 90 00  ..............
>>>
```

File: test/regression.uts (hunk 1 above)
Comment (gpotter2) [reply test/regression.uts]: See my comment below. I however just noticed that it should have been capitalized letters. Just fixed that (`%02x` ==> `%02X`)

**Pull request merged**